server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -354,8 +354,6 @@ private static boolean searchWithCollector(SearchContext searchContext, ContextI
                 throw new QueryPhaseExecutionException(searchContext.shardTarget(), "Time exceeded");
             }
             queryResult.searchTimedOut(true);
-        } finally {
-            searchContext.clearReleasables(SearchContext.Lifetime.COLLECTION);
```

Comment:
I am wondering why we had this statement before?  I don't see anywhere in the code that we add a releasable with a scope of `Lifetime.COLLECTION`.

Reply:
Good question :). The lifetime of a SearchContext now is within a single phase or Lifetime.COLLECTION as before. With that, we do not have to track releasables with the lifetime COLLECTION separately as we can just add them to the releasables of a SearchContext.